### DISCUSSION TITLE
Kalman Filter in non-linear models

### DISCUSSION THREAD

**Discussion by pfjulio** [IRIS-Solutions-Team/IRIS-Toolbox#343] [General]
Hi Jaromir,

Is is possible to get the smoothed variables (filter the model with the KalmanFilter) in a non-linear model with the left and right hand sides of all equations being exatly equal in each and every period (instead of having smoothed variables based on a first-order approximation of the model, which generates discrepancies between the LHS and the RHS)?

Many thanks

**Comment by jaromir-benes** [comment]
Hi - no, there is no option to run the kalman filter this way. If you have
the same number of shocks as the number of observed variables, you can use
nonlinear simulation with an exogenize/endogenize plan. Best, Jaromir
